lib/spack/spack/cmd/create.py

```diff
@@ -249,43 +264,53 @@ def install(self, spec, prefix):
                    prefix, self.stage.archive_file))"""
 
     def __init__(self, name, *args):
-        name = 'octave-{0}'.format(name)
-        super(OctaveGuess, self).__init__(name, *args)
+        if not name.startswith('octave-'):
+            name = 'octave-{0}'.format(name)
+
+        super(OctavePackageTemplate, self).__init__(name, *args)
+
+
+templates = {
+    'autotools': AutotoolsPackageTemplate,
+    'cmake':     CMakePackageTemplate,
+    'scons':     SconsPackageTemplate,
+    'bazel':     BazelPackageTemplate,
+    'python':    PythonPackageTemplate,
+    'R':         RPackageTemplate,
+    'octave':    OctavePackageTemplate,
+    'default':   PackageTemplate
+}
 
 
 def setup_parser(subparser):
-    subparser.add_argument('url', nargs='?', help="url of package archive")
+    subparser.add_argument(
+        'url', nargs='?',
+        help="url of package archive")
     subparser.add_argument(
         '--keep-stage', action='store_true',
-        help="Don't clean up staging area when command completes.")
+        help="don't clean up staging area when command completes")
+    subparser.add_argument(
+        '-n', '--name',
+        help="name of the package to create")
     subparser.add_argument(
-        '-n', '--name', dest='alternate_name', default=None, metavar='NAME',
-        help="Override the autodetected name for the created package.")
+        '-t', '--template', metavar='TEMPLATE', choices=templates.keys(),
+        help="build system template to use. options: %(choices)s")
     subparser.add_argument(
-        '-r', '--repo', default=None,
-        help="Path to a repository where the package should be created.")
+        '-r', '--repo',
+        help="path to a repository where the package should be created")
     subparser.add_argument(
         '-N', '--namespace',
-        help="Specify a namespace for the package. Must be the namespace of "
-        "a repository registered with Spack.")
+        help="namespace for the package")
```

Comment:
Can someone test `--repo` and `--namespace` for me? I don't use either, so I can't really test whether or not the behavior has changed.